argon2/argon2.go

```diff
@@ -100,6 +100,9 @@ func IDKey(password, salt []byte, time, memory uint32, threads uint8, keyLen uin
 }
 
 func deriveKey(mode int, password, salt, secret, data []byte, time, memory uint32, threads uint8, keyLen uint32) []byte {
+	if keyLen == 0 {
+		return []byte{}
+	}
```

Comment:
I think it's slightly more technically correct to put this after the time/threads checks directly below

Reply:
It seems you are right. Done